File: analytics/metrics-platform/src/test/java/org/wikimedia/metrics_platform/config/CurationFilterFixtures.java
package org.wikimedia.metrics_platform.config;

import java.util.Arrays;

import org.wikimedia.metrics_platform.config.curation.CollectionCurationRules;
import org.wikimedia.metrics_platform.config.curation.CurationRules;

public final class CurationFilterFixtures {

    private CurationFilterFixtures() {
        // Utility class - should never be instantiated
    }

    public static CurationFilter getCurationFilter() {
        return CurationFilter.builder()
                .pageTitleRules(CurationRules.<String>builder().isEquals("Test").build())
                .performerGroupsRules(
                        CollectionCurationRules.<String>builder()
                                .doesNotContain("sysop")
                                .containsAny(Arrays.asList("steward", "bureaucrat"))
                                .build()
                )
                .build();
    }
}

File: analytics/testkitchen/src/test/java/org/wikimedia/testkitchen/config/CurationFilterFixtures.kt
package org.wikimedia.testkitchen.config

import org.wikimedia.testkitchen.config.curation.CollectionCurationRules
import org.wikimedia.testkitchen.config.curation.CurationRules

object CurationFilterFixtures {
    val curationFilter: CurationFilter
        get() = CurationFilter().also {
            it.pageTitleRules = CurationRules<String>().also { rules ->
                rules.isEquals = "Test"
            }
            it.performerGroupsRules = CollectionCurationRules<String>().also { rules ->
                rules.doesNotContain = "sysop"
                rules.containsAny = listOf("steward", "bureaucrat")
            }
        }
}
